src/elements/content-sidebar/versions/VersionsSidebarContainer.js

```diff
@@ -67,11 +69,17 @@ class VersionsSidebarContainer extends React.Component<Props, State> {
     window: any = window;
 
     componentDidMount() {
-        this.fetchData().catch(this.handleFetchError);
+        this.initialize();
+        this.fetchData();
     }
 
-    componentDidUpdate({ versionId: prevVersionId }: Props) {
-        const { versionId } = this.props;
+    componentDidUpdate({ fileId: prevFileId, versionId: prevVersionId }: Props) {
+        const { fileId, versionId } = this.props;
+
+        if (fileId !== prevFileId) {
+            this.initialize();
```

Comment:
This should be the same instance of the APIFactory so there isn't any cleanup required right?

Reply:
Yeah, it's just meant to update the helper's `fileId` property. I could make a setter for it rather than re-instantiating the class. Thoughts?